sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -192,17 +185,20 @@ private set
             PrefetchCount = _options.PrefetchCount;
             MaxAutoLockRenewalDuration = _options.MaxAutoLockRenewalDuration;
             MaxConcurrentCalls = _options.MaxConcurrentCalls;
-            if (MaxConcurrentCalls == 0)
-            {
-                MaxConcurrentCalls = isSessionEntity ? DefaultMaxConcurrentSessions : DefaultMaxConcurrentCalls;
-            }
-            MessageHandlerSemaphore = new SemaphoreSlim(
-                MaxConcurrentCalls,
-                MaxConcurrentCalls);
+            MaxConcurrentSessions = maxConcurrentSessions;
+            MaxConcurrentCallsPerSession = maxConcurrentCallsPerSession;
+
+            int maxCalls = isSessionEntity ?
+                MaxConcurrentSessions * MaxConcurrentCallsPerSession :
+                MaxConcurrentCalls;
 
+            MessageHandlerSemaphore = new SemaphoreSlim(
+                maxCalls,
+                maxCalls);
+            var maxAcceptSessions = Math.Min(maxCalls, 2 * Environment.ProcessorCount);
```

Comment:
I know it was in the previous code above, but why is this `2 * Environment.ProcessorCount`? If the idea is to get the processor thread count, doubling the processor count would only apply for CPUs with hyperthreading enabled. Not all systems, like VMs, would have hyperthreading enabled.

Reply:
This was carried forward from Track 1, and is meant to be a simple heuristic to avoid attempting to "accept" an excessive number of sessions simultaneously. Once the session is accepted it will stay open until we run out of messages, whereas the lock will be released from the semaphore allowing an additional session to be accepted. So in practice the total number of sessions that can be open at a point in time would be MaxConcurrentSessions, but we will restrict concurrent act of accepting them to this `maxAcceptSessions`. I assume that the origin of the the `2 * Environment.Processor` value is based on 2 threads per processor, but I don't think this condition is essential for this to be useful.